### ISSUE TITLE
RecordBatch might have logical row mapping on physical arrays

### ISSUE THREAD

**Issue by viirya** [apache/datafusion-comet#974]
### Describe the bug

This is related to #973. After applying the fix, the test we run locally with Iceberg table with deleted rows fails on incorrect query result.

It is because Iceberg doesn't actually delete the rows from the arrays but stores a row mapping in arrays that is used to skip deleted rows when iterating rows in a columnar batch.

However, once we export the underlying Arrow record batch of a columnar batch to native side. The record batch is totally physical without the logical row mapping info. That's said, the deleted rows will occur in the query.

When exporting a record batch from Iceberg record batch, we need to export the row mapping together.

**Comment by andygrove** [comment]
@parthchandra @huaxingao is this still a valid issue that we need to fix?

**Comment by huaxingao** [comment]
Yes, we still need this

**Comment by parthchandra** [comment]
Yes. Comet does not support deleted rows in Iceberg for this reason which reduces its usefulness somewhat.

**Comment by andygrove** [comment]
@parthchandra @hsiang-c  I think that this can be closed now. Please reopen if I am mistaken.
